lib/active_admin/resource/attributes.rb

```diff
@@ -0,0 +1,44 @@
+module ActiveAdmin
+
+  class Resource
+    module Attributes
+
+      def default_attributes
+        resource_class.columns.each_with_object({}) do |c, attrs|
+          unless reject_col?(c)
```

Comment:
Ok, I think I get it now.

* _Columns ended with `_id`._ Rails's `content_columns` excludes these, but we don't want to do that because we want to show those in our tables and CSV exports when they are not linked to an association . So we switch from AR `content_columns` to just `columns` and don't filter out these columns. :+1: for that change in behaviour. 

* _Columns ended with `_count`_. By the same argument, I'd say we want to have these columns as well, since they are just regular columns not linked to any association. This would be a change in behavior, just like adding `_id`-ended columns. :+1: for this change in behavior as well. I think it's better behavior and it saves us from duplicating AR functionality. 

* _Primary key column_. In this case, my suggestion doesn't change the behavior but it leads to simpler code because we don't duplicate AR functionality either. So :+1: to simplifying that.

Since this PR is tagged as "refactoring", I'd suggest to try not duplicate AR functionality if we don't need to. So I would:

* Drop `reject_col?` method.
* Drop `counter_cache_col?` method.
* Drop `primary_col?` method.
* Drop https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/csv_builder.rb#L24 line.
* Drop https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/views/index_as_table.rb#L235 line.